api/src/main/java/org/openmrs/api/db/DAOException.java

```diff
@@ -33,4 +34,14 @@ public DAOException(Throwable cause) {
 		super(cause);
 	}
 	
+	/**
+	 * Constructor to give the end user a helpful message that relates to why this error occurred.
+	 *
+	 * @param messageKey message code to retrieve
+	 * @param parameters message parameters
+	 */
+	public DAOException(String messageKey, Object[] parameters) {
```

Comment:
@wisogetnid this ticket you are working on is only about adding tests, can you please just add your tests to this PR.

Regarding the change of the exception I am unsure if we want to add the context to this exception. @djazayeri what do you say?